### PR TITLE
Upgrade to 0.0.24b

### PR DESCRIPTION
**Pull request by barthalion**
Opening a new PR because I have no write access to @corecontingency's fork. Feel free to cherry-pick the latest commit if it works and merging yours instead.

### CODE REVIEW

**Comment by barthalion** [comment]
The actual error message is this:

```
error: failed to parse manifest at `/run/build/0ad/libraries/source/spidermonkey/mozjs-78.6.0/Cargo.toml`

Caused by:
  dependency (nix) specification is ambiguous. Only one of `branch`, `tag` or `rev` is allowed.
```

**Comment by barthalion** [comment]
Sigh, looks like Rust dependencies are not vendored:

```
warning: spurious network error (2 tries remaining): failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
warning: spurious network error (1 tries remaining): failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
error: failed to resolve patches for `https://github.com/rust-lang/crates.io-index`

Caused by:
  failed to load source for dependency `nix`

Caused by:
  Unable to update https://github.com/shravanrn/nix/?rev=4af6c367603869a30fddb5ffb0aba2b9477ba92e

Caused by:
  failed to clone into: /var/home/bpiotrowski/.cargo/git/db/nix-bcf9654e7f420cf5

Caused by:
  network failure seems to have happened
  if a proxy or similar is necessary `net.git-fetch-with-cli` may help here
  https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli

Caused by:
  failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
```

Might be easier to build mozjs separately. Will continue tomorrow.

**Comment by barthalion** [comment]
```
ld.lld: error: undefined hidden symbol: encoding_mem_convert_latin1_to_utf16
>>> referenced by Latin1.h:256 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:256)
>>>               ../Unified_cpp_js_src11.o:(js::StringBuffer::inflateChars())
>>> referenced by Latin1.h:256 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:256)
>>>               ../Unified_cpp_js_src11.o:(js::InflateString(JSContext*, char const*, unsigned long))
>>> referenced by Latin1.h:256 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:256)
>>>               ../Unified_cpp_js_src23.o:(void js::CopyChars<char16_t>(char16_t*, JSLinearString const&))
>>> referenced by Latin1.h:256 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:256)
>>>               ../Unified_cpp_js_src23.o:(js::CopyAndInflateChars(char16_t*, unsigned char const*, unsigned long))
>>> referenced by Latin1.h:256 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:256)
>>>               ../Unified_cpp_js_src23.o:(JS::AutoStableStringChars::copyAndInflateLatin1Chars(JSContext*, JS::Handle<JSLinearString*>))
>>> referenced by Latin1.h:256 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:256)
>>>               ../Unified_cpp_js_src7.o:(JS_DecodeBytes(JSContext*, char const*, unsigned long, char16_t*, unsigned long*))
>>> referenced by Latin1.h:256 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:256)
>>>               ../Unified_cpp_js_src7.o:(JS_DecodeBytes(JSContext*, char const*, unsigned long, char16_t*, unsigned long*))
>>> referenced by Latin1.h:256 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:256)
>>>               ../gc/Unified_cpp_js_src_gc0.o:(JS::GCDescription::formatSliceMessage(JSContext*) const)
>>> referenced by Latin1.h:256 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:256)
>>>               ../gc/Unified_cpp_js_src_gc0.o:(JS::GCDescription::formatSummaryMessage(JSContext*) const)
>>> referenced by Latin1.h:256 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:256)
>>>               ../gc/Unified_cpp_js_src_gc0.o:(JS::GCDescription::formatJSONTelemetry(JSContext*, unsigned long) const)

ld.lld: error: undefined hidden symbol: encoding_mem_convert_utf16_to_latin1_lossy
>>> referenced by Latin1.h:171 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:171)
>>>               ../Unified_cpp_js_src14.o:(JS::LossyTwoByteCharsToNewLatin1CharsZ(JSContext*, mozilla::Range<char16_t const>))
>>> referenced by Latin1.h:171 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:171)
>>>               ../Unified_cpp_js_src23.o:(void js::CopyChars<unsigned char>(unsigned char*, JSLinearString const&))
>>> referenced by Latin1.h:171 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:171)
>>>               ../Unified_cpp_js_src23.o:(JSLinearString* NewStringDeflated<(js::AllowGC)1>(JSContext*, char16_t const*, unsigned long))
>>> referenced by Latin1.h:171 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:171)
>>>               ../Unified_cpp_js_src23.o:(JSLinearString* NewStringDeflated<(js::AllowGC)1>(JSContext*, char16_t const*, unsigned long))
>>> referenced by Latin1.h:171 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:171)
>>>               ../Unified_cpp_js_src23.o:(JSLinearString* NewStringDeflated<(js::AllowGC)0>(JSContext*, char16_t const*, unsigned long))
>>> referenced by Latin1.h:171 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:171)
>>>               ../Unified_cpp_js_src23.o:(JSLinearString* NewStringDeflated<(js::AllowGC)0>(JSContext*, char16_t const*, unsigned long))
>>> referenced by Latin1.h:171 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:171)
>>>               ../Unified_cpp_js_src23.o:(js::NewMaybeExternalString(JSContext*, char16_t const*, unsigned long, JSExternalStringCallbacks const*, bool*))

ld.lld: error: undefined hidden symbol: encoding_mem_convert_utf16_to_utf8_partial
>>> referenced by Utf8.h:327 (/run/build/mozjs/build/dist/include/mozilla/Utf8.h:327)
>>>               ../Unified_cpp_js_src14.o:(JS::DeflateStringToUTF8Buffer(JSLinearString*, mozilla::Span<char, 18446744073709551615ul>))
>>> referenced by Utf8.h:327 (/run/build/mozjs/build/dist/include/mozilla/Utf8.h:327)
>>>               ../Unified_cpp_js_src14.o:(JS::UTF8CharsZ JS::CharsToNewUTF8CharsZ<char16_t>(JSContext*, mozilla::Range<char16_t>))
>>> referenced by Utf8.h:327 (/run/build/mozjs/build/dist/include/mozilla/Utf8.h:327)
>>>               ../Unified_cpp_js_src14.o:(JS::UTF8CharsZ JS::CharsToNewUTF8CharsZ<char16_t const>(JSContext*, mozilla::Range<char16_t const>))
>>> referenced by Utf8.h:327 (/run/build/mozjs/build/dist/include/mozilla/Utf8.h:327)
>>>               ../Unified_cpp_js_src23.o:(JSString::encodeUTF8Partial(JS::AutoRequireNoGC const&, mozilla::Span<char, 18446744073709551615ul>) const)

ld.lld: error: undefined hidden symbol: encoding_mem_convert_latin1_to_utf8_partial
>>> referenced by Latin1.h:227 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:227)
>>>               ../Unified_cpp_js_src14.o:(JS::DeflateStringToUTF8Buffer(JSLinearString*, mozilla::Span<char, 18446744073709551615ul>))
>>> referenced by Latin1.h:227 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:227)
>>>               ../Unified_cpp_js_src14.o:(JS::UTF8CharsZ JS::CharsToNewUTF8CharsZ<unsigned char>(JSContext*, mozilla::Range<unsigned char>))
>>> referenced by Latin1.h:227 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:227)
>>>               ../Unified_cpp_js_src14.o:(JS::UTF8CharsZ JS::CharsToNewUTF8CharsZ<unsigned char const>(JSContext*, mozilla::Range<unsigned char const>))
>>> referenced by Latin1.h:227 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:227)
>>>               ../Unified_cpp_js_src23.o:(JSString::encodeUTF8Partial(JS::AutoRequireNoGC const&, mozilla::Span<char, 18446744073709551615ul>) const)

ld.lld: error: undefined hidden symbol: encoding_ascii_valid_up_to
>>> referenced by TextUtils.h:136 (/run/build/mozjs/build/dist/include/mozilla/TextUtils.h:136)
>>>               ../Unified_cpp_js_src14.o:(JS::FindSmallestEncoding(JS::UTF8Chars))

ld.lld: error: undefined hidden symbol: encoding_mem_is_utf8_latin1
>>> referenced by Latin1.h:102 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:102)
>>>               ../Unified_cpp_js_src14.o:(JS::FindSmallestEncoding(JS::UTF8Chars))

ld.lld: error: undefined hidden symbol: encoding_mem_is_ascii
>>> referenced by TextUtils.h:72 (/run/build/mozjs/build/dist/include/mozilla/TextUtils.h:72)
>>>               ../Unified_cpp_js_src14.o:(JS::StringIsASCII(mozilla::Span<char const, 18446744073709551615ul>))
>>> referenced by TextUtils.h:72 (/run/build/mozjs/build/dist/include/mozilla/TextUtils.h:72)
>>>               ../Unified_cpp_js_src23.o:(js::StringIsAscii(JSLinearString*))

ld.lld: error: undefined hidden symbol: install_rust_panic_hook
>>> referenced by Initialization.cpp:131 (/run/build/mozjs/js/src/vm/Initialization.cpp:131)
>>>               ../Unified_cpp_js_src17.o:(JS::detail::InitWithFailureDiagnostic(bool))

ld.lld: error: undefined hidden symbol: encoding_mem_convert_utf16_to_utf8
>>> referenced by Utf8.h:300 (/run/build/mozjs/build/dist/include/mozilla/Utf8.h:300)
>>>               ../Unified_cpp_js_src18.o:(void PrintSingleError<JSErrorReport>(JSContext*, _IO_FILE*, JS::ConstUTF8CharsZ, JSErrorReport*, PrintErrorKind))

ld.lld: error: undefined hidden symbol: encoding_utf8_valid_up_to
>>> referenced by Utf8.h:271 (/run/build/mozjs/build/dist/include/mozilla/Utf8.h:271)
>>>               ../Unified_cpp_js_src19.o:(js::ScriptSourceObject::initFromOptions(JSContext*, JS::Handle<js::ScriptSourceObject*>, JS::ReadOnlyCompileOptions const&))
>>> referenced by Utf8.h:271 (/run/build/mozjs/build/dist/include/mozilla/Utf8.h:271)
>>>               ../wasm/Unified_cpp_js_src_wasm3.o:(DecodeName(js::wasm::Decoder&))

ld.lld: error: undefined hidden symbol: encoding_mem_is_basic_latin
>>> referenced by TextUtils.h:105 (/run/build/mozjs/build/dist/include/mozilla/TextUtils.h:105)
>>>               ../Unified_cpp_js_src23.o:(js::StringIsAscii(JSLinearString*))

ld.lld: error: undefined hidden symbol: encoding_mem_is_utf16_latin1
>>> referenced by Latin1.h:89 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:89)
>>>               ../Unified_cpp_js_src23.o:(JSLinearString* js::NewString<char16_t>(JSContext*, mozilla::UniquePtr<char16_t [], JS::FreePolicy>, unsigned long))
>>> referenced by Latin1.h:89 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:89)
>>>               ../Unified_cpp_js_src23.o:(JSLinearString* js::NewString<(js::AllowGC)1, char16_t>(JSContext*, mozilla::UniquePtr<char16_t [], JS::FreePolicy>, unsigned long))
>>> referenced by Latin1.h:89 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:89)
>>>               ../Unified_cpp_js_src23.o:(JSLinearString* js::NewStringCopyN<(js::AllowGC)1, char16_t>(JSContext*, char16_t const*, unsigned long))
>>> referenced by Latin1.h:89 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:89)
>>>               ../Unified_cpp_js_src23.o:(JSLinearString* js::NewStringCopyN<(js::AllowGC)0, char16_t>(JSContext*, char16_t const*, unsigned long))
>>> referenced by Latin1.h:89 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:89)
>>>               ../Unified_cpp_js_src23.o:(js::NewMaybeExternalString(JSContext*, char16_t const*, unsigned long, JSExternalStringCallbacks const*, bool*))
>>> referenced by Latin1.h:89 (/run/build/mozjs/build/dist/include/mozilla/Latin1.h:89)
>>>               ../Unified_cpp_js_src23.o:(bool FillWithRepresentatives<JSString::fillWithRepresentatives(JSContext*, JS::Handle<js::ArrayObject*>)::$_4, char16_t>(JSContext*, JS::Handle<js::ArrayObject*>, unsigned int*, char16_t const*, unsigned long, unsigned long, JSString::fillWithRepresentatives(JSContext*, JS::Handle<js::ArrayObject*>)::$_4 const&))
clang-10: error: linker command failed with exit code 1 (use -v to see invocation)
```

This is where I'm stuck now.

File: com.play0ad.zeroad.yaml [hunk 1]
```diff
@@ -36,28 +36,67 @@ modules:
...
+  config-opts:
+    - --program-suffix=2.13
+  cleanup:
+  - "*"
+  sources:
+  - type: archive
+    url: https://ftp.gnu.org/gnu/autoconf/autoconf-2.13.tar.gz
+    sha256: f0611136bee505811e9ca11ca7ac188ef5323a8e2ef19cffd3edb3cf08fd791e
+
+- name: mozjs
```
Comment (alatiera) [inline com.play0ad.zeroad.yaml:77]: We have mozjs 78 in fd.o 20.08. I think you can avoid building it.

Comment (alatiera) [reply com.play0ad.zeroad.yaml:77]: Nevermind me, its not part of the runtime.

**Pull request merged**